### DISCUSSION TITLE
Can't Upload Callsign DB

### DISCUSSION THREAD

**Discussion by dkaye** [hmatuschek/qdmr#358] [Q&A]
Using a Baofeng DM-1701 and QDMR 0.11.2 on Mac OS. Via the GUI, the progress bar looks like the database is uploading, but I can't find it on the radio. I then tried the command-line interface:

      dmrconf --id=3196950  write-db

It runs to 100%, then I get these errors:

     ERROR in lib/dfu_libusb.cc@238: Cannot get status: Pipe error.
     ERROR in lib/dfu_libusb.cc@200: Cannot write to device: Pipe error.
     ERROR in lib/dfu_libusb.cc@431: Cannot leave DFU mode.

What am I doing wrong?

**Comment by hmatuschek** [comment]
Probably nothing, some radios reboot too fast after the data has been written. Then, qdmr cannot close the connection to the device properly and you get some error messages. This should not be critical. 

Concerning the non-functional calls-sign DB: Make sure, that you have the correct firmware installed. For the TyT-like devices, there are usually two variants. One allows for recordings, the other uses the space for the call-sign DB. The one is usually labeled "REC" the other "CSV".

**Comment by allesand** [comment]
According to https://github.com/hmatuschek/qdmr/discussions/376 / https://github.com/hmatuschek/qdmr/issues/377 , uploading the callsign DB to a DM-1701 should work?

**Comment by skipper-henrik** [comment]
for me it works. how much is the maximal number?

**Reply by allesand** [reply]
QDMR knows how many entries fit in the radio (https://dm3mat.darc.de/qdmr/manual/ch02s11.html) so you do not need to set that manually. Shold be 120k on the DM-1701?

**Reply by allesand** [reply]
Bonus question: @eltnap can you confirm that callsign display works correctly on the DM-1701? There is an open issue https://github.com/hmatuschek/qdmr/issues/296 so your confirmation would help.
